src/Physics/Plugins/ammoJSPlugin.ts

```diff
@@ -66,11 +70,22 @@ export class AmmoJSPlugin implements IPhysicsEnginePlugin {
         }
 
         // Initialize the physics world
-        this._collisionConfiguration = new this.bjsAMMO.btDefaultCollisionConfiguration();
-        this._dispatcher = new this.bjsAMMO.btCollisionDispatcher(this._collisionConfiguration);
-        this._overlappingPairCache = new this.bjsAMMO.btDbvtBroadphase();
-        this._solver = new this.bjsAMMO.btSequentialImpulseConstraintSolver();
-        this.world = new this.bjsAMMO.btDiscreteDynamicsWorld(this._dispatcher, this._overlappingPairCache, this._solver, this._collisionConfiguration);
+        if (this._includeSoftBodies) {
+            this._collisionConfiguration = new this.bjsAMMO.btSoftBodyRigidBodyCollisionConfiguration();

```

Comment:
I am worried about this._collisionConfiguration and this.world their type in this file was set to any as ammo does not provide typings for them but now when softbodies are enabled these fields can be of different types.

This makes writing code like this needed
```
        this.world.setGravity(this._tmpAmmoVectorA);
        if (this._includeSoftBodies) {
            this.world.getWorldInfo().set_m_gravity(this._tmpAmmoVectorA);
        }
```
Ideally typescript typings should be used to differentiate them. Maybe there should be 2 separate variables.
(eg.  this.world and this.softBodyWorld) that way we could look to add ammojs typings in the future

Reply:
Problem here is that when soft bodies are included rigid bodies are also included in the same world so cannot have two separate variables for the world.  I am working at the boundaries of my Typescript knowledge talking about inheritance but if there was an extension of AmmoJSPlugin to AmmoJSPlugin would the mean you could have 

in AmmoJSPlugin
```
public setGravity(gravity: Vector3): void {
        this._tmpAmmoVectorA.setValue(gravity.x, gravity.y, gravity.z);
        this.world.setGravity(this._tmpAmmoVectorA);
    }
```
and in AmmoSoftJSPlugin
```
public setGravity(gravity: Vector3): void {
        this._tmpAmmoVectorA.setValue(gravity.x, gravity.y, gravity.z);
        this.world.setGravity(this._tmpAmmoVectorA);
        this.world.getWorldInfo().set_m_gravity(this._tmpAmmoVectorA);
    }
```
In fact you would leave AmmoJSPlugin as it currently is and in AmmoSoftJSPlugin where I have got

```
if (_includeSoftBodies) {
      do A
}
else {
      do B
}
```

you do 

```
do A
```

Reply:
I'm worries that this could be complex from an user standpoint. We should not multiplicate the plugins.

Further more does that mean that soft and rigid bodies cannot be used simultaneously?

Reply:
AmmoJSPlugin would be for rigid bodies only, whereas AmmoSoftJSPlugin would be for both soft and rigid bodies. (Could call it AmmoSoftandRigidJSPlugin but too long). I am quite happy to leave it as one plugin along the lines I am doing now. However I cannot then see away around having code of the form

```
 this.world.setGravity(this._tmpAmmoVectorA);
if (this._includeSoftBodies) {
     this.world.getWorldInfo().set_m_gravity(this._tmpAmmoVectorA);
 }
```
which Trevor was concerned about.

Currently I am using two flags to deal with soft physics. The _includeSoftBodies flag is for setting the Ammo world parameters either false for a rigid world or true for a soft and rigid world. The soft flag is a property of impostors and determines how they are treated.

Reply:
I let @TrevorDev decide here but I'm more on the one single plugin side ;)

Reply:
I agree that having another plugin is undesirable. After looking at this a bit more I think the way you have it now is probably fine and if we add typings this can exist as a union type.

Reply:
I unresolved this because I thought it depended on the answer to the above conversation but it doesn't. What ever the answer this is resolved. Apologies.